custom-node/genesis/src/main.rs

```diff
@@ -0,0 +1,27 @@
+use anyhow::Result;
+use std::path::Path;
+use structopt::StructOpt;
+use diem_config::config::NodeConfig;
+
+#[derive(StructOpt)]
+#[structopt(
+    name = "custom-node",
+    about = "Builds a WriteSet transaction to install the custom modules and starts a node",
+    rename_all = "kebab-case"
+)]
+pub struct CustomFramework {
+    /// Directory where the node config will be generated. Must not already exist
+    #[structopt(long = "node-config-dir")]
+    node_config_dir: String,
+}
+/// Generate a node config under `args.node_config_dir`
+fn main() -> Result<()> {
```

Comment:
Why can't we just use the diem-node binary directly and add a flag which points to the necessary stuff?

Reply:
What this code does is essentially generating the "flag", where the flag is a validator config, and then starting a diem node with that flag. 

If I delete the second last line in main function (`diem_node::start(&validator_config, None);`), all this code does is generating the node config to some file. I can then start a diem-node binary with the newly generated node config. All of that will have the same effect as what the main function is doing now, right?

Reply:
Yes I think that would make a lot more sense, have code which can be used to generate the custom modules but then just add a flag to the existing binary to be able to load the custom modules as apart of genesis.